### PR TITLE
 [SDK-1684] : Web view Enhancements 

### PR DESCRIPTION
**Pull request by root-ansh-ct**
[SDK-1684] : supporting onUserLogin,increment and decrement apis via webview

[SDK-1684]: https://wizrocket.atlassian.net/browse/SDK-1684?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CTWebInterfaceTest.kt [hunk 1]
```diff
@@ -39,6 +39,43 @@ class CTWebInterfaceTest : BaseTestCase() {
...
+
+
+        // when ctApi is not null, calling this function will call ctApi's internal function
+        ctApi = CleverTapAPI.getDefaultInstance(application)
+        val ctMock = Mockito.mock(CleverTapAPI::class.java)
+        ctWebInterface = CTWebInterface(ctMock)
+        ctWebInterface.incrementValue("key2",2.5)
+        Mockito.verify(ctMock,Mockito.times(1))?.incrementValue("key2",2.5)
+    }
+
```
Comment (root-ansh-ct) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/CTWebInterfaceTest.kt:58]: add unit tests for extreme  min/max ranges , eg Long.maxValue, Double.minValue, etc

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CTWebInterfaceTest.kt [hunk 2]
```diff
@@ -39,6 +39,43 @@ class CTWebInterfaceTest : BaseTestCase() {
...
+        ctWebInterface.incrementValue("key",1.5)
+
+
+        // when ctApi is not null, calling this function will call ctApi's internal function
+        ctApi = CleverTapAPI.getDefaultInstance(application)
+        val ctMock = Mockito.mock(CleverTapAPI::class.java)
+        ctWebInterface = CTWebInterface(ctMock)
+        ctWebInterface.incrementValue("key2",2.5)
+        Mockito.verify(ctMock,Mockito.times(1))?.incrementValue("key2",2.5)
+    }
```
Comment (root-ansh-ct) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/CTWebInterfaceTest.kt:57]: add tests for extreme ranges Int.MAX_VALUE , Double.MIN_VALUE, etc (for int/long/double/float)

File: sample/src/main/assets/sampleHTMLCode.html [hunk 3]
```diff
@@ -42,6 +48,17 @@
...
+        function invokeOnUserLogin() {
+            var props = {foo: 'zyx', lang: 'German'};
+            CleverTap.onUserLogin(JSON.stringify(props))
+        }
+        function invokeIncrementValue() {
+            CleverTap.incrementValue('cars',2)
+        }
+        function invokeDecrementValue() {
+            CleverTap.decrementValue('cars',2)
+
```
Comment (root-ansh-ct) [inline sample/src/main/assets/sampleHTMLCode.html:60]: add demo for  long/double/float

**Review by piyush-kukadiya — APPROVED** [review]
LGTM

**Pull request merged**